blueocean-rest-impl/src/main/java/io/jenkins/blueocean/service/embedded/rest/RunSearch.java

```diff
@@ -209,6 +198,7 @@ private static List<BlueRun> collectRunsParallel(Iterator<? extends Run> runIter
             executorService.shutdownNow();
         }
         LOGGER.debug("runs found:{}", runs.size());
+        runs.sort(Comparator.comparing((BlueRun r) -> Integer.valueOf(r.getId())).reversed());
```

Comment:
Curious that the original `Run.number` is not retained. The `id` is just a stringified version.